src/App.vue

```diff
@@ -166,10 +169,30 @@ import Alerter from './components/widgets/Alerter.vue'
 import { datalogger } from './libs/sensors-logging'
 import { useMainVehicleStore } from './stores/mainVehicle'
 import { useWidgetManagerStore } from './stores/widgetManager'
-
 const widgetStore = useWidgetManagerStore()
 const vehicleStore = useMainVehicleStore()
 
+onMounted(async () => {
+  const baseUrl = `${window.location.protocol}//${window.location.host}${window.location.pathname}`
+  const normalizedBaseUrl = baseUrl.endsWith('/') ? baseUrl : `${baseUrl}/`
+
+  try {
+    const configUrl = new URL('config.json', normalizedBaseUrl)
+    const configResponse = await fetch(configUrl.toString())
+    if (!configResponse.ok) {
+      throw new Error(`Failed to fetch config.json: ${configResponse.status}`)
+    }
+    const config = await configResponse.json()
+    console.log(config)
+
```

Comment:
I probably need to add a check for if the uri is empty here

Reply:
Yes. I think it's totally fine, as long as we don't enforce it. Cockpit should work as a BlueOS extension as well as a standalone frontend, or an Electron application (although this is not being maintained right now, so it's not totally needed).